web/react/components/mention_list.jsx

```diff
@@ -23,6 +23,11 @@ module.exports = React.createClass({
                 }
             }
         );
+        $(document).click(function() {
```

Comment:
Here we get the DOM object of the entire page, or document, and say "when a click event is registered on the document, trigger whatever function is passed as an argument".